app/src/main/java/com/odysee/app/ui/golive/GoLiveFragment.java

```diff
@@ -0,0 +1,692 @@
+                        activity.getWindow().clearFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
+                    }
+                    connection.close();
+                    isStreaming = false;
+                    startingStream = false;
+                    buttonToggleStreaming.setText(R.string.start_streaming);
+                }
+            }
+        });
+
+        buttonSwitchCamera.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                cameraSource.switchCamera();
+            }
+        });
+
+        return root;
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+
+        if (screenTurnedOn) {
+            Activity activity = getActivity();
+            if (activity instanceof MainActivity) {
+                activity.onBackPressed();
+
+                View view = activity.findViewById(R.id.content_main);
+                Snackbar snackbar = Snackbar.make(view, R.string.stream_stopped_went_to_home_reason, Snackbar.LENGTH_LONG);
+                TextView snackbarText = snackbar.getView().findViewById(R.id.snackbar_text);
+                snackbarText.setMaxLines(Integer.MAX_VALUE);
+                snackbar.show();
+            }
+            return;
+        }
+
+        Context context = getContext();
+        if (context instanceof MainActivity) {
+            MainActivity activity = (MainActivity) context;
+            LbryAnalytics.setCurrentScreen(activity, "Go Live", "GoLive");
+            activity.setBackPressInterceptor(this);
+            activity.addCameraPermissionListener(this);
+            activity.addStoragePermissionListener(this);
+            activity.addFilePickerListener(this);
+        }
+        checkCameraPermissionAndOpenCameraSource();
+        fetchChannels();
+    }
+
+    @Override
+    public void onStop() {
+        connection.close();
+        Context context = getContext();
+        if (context instanceof MainActivity) {
+            MainActivity activity = (MainActivity) context;
+            activity.removeCameraPermissionListener(this);
+            activity.removeStoragePermissionListener(this);
+            if (!MainActivity.startingFilePickerActivity) {
+                activity.removeFilePickerListener(this);
+            }
+        }
+        super.onStop();
+    }
+
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+        connection.dispose();
+        Activity activity = getActivity();
+        if (activity != null && screenOnReceiver != null) {
+            activity.unregisterReceiver(screenOnReceiver);
+        }
+    }
+
+    @Override
+    public boolean onBackPressed() {
+        if (didShowCloseWarning) {
+            return false;
+        }
+
+        /*
+        FIXME: Toolbar back button gets hidden when pressed
+        Therefore answering "No" to the dialog leaves the toolbar without a back button
+        */
```

Comment:
Any suggestions how to fix this?

I looked and it's not being caused by any calls to `setDisplayHomeAsUpEnabled` in the Odysee codebase.

Reply:
Absolutely not your fault. I have already seen this problem with other fragments, like the channel manager fragment.